storage/innobase/include/dict0mem.h

```diff
@@ -618,7 +618,7 @@ struct dict_col_t{
 		ut_ad(mtype == DATA_INT || mtype == DATA_FIXBINARY);
 		return mtype == DATA_INT;
 	}
-	/** @return whether this is system versioned */
+	/** @return whether this is versioned user field */
 	bool is_versioned() const { return !(~prtype & DATA_VERSIONED); }
```

Comment:
The comment is confusing index fields with columns. The member function is for `dict_col_t`, that is, representing the columns of a table. In InnoDB, ‘field’ refers to something that exists in an index. The fields of an index entry can be copies of full columns or prefixes of string columns.

The term ‘user field’ was introduced with `dict_index_t::first_user_field()`. Does this comment change now suggest that `PRIMARY KEY` columns cannot be versioned?

If there is some rule that restricts the choice of versioned columns, I think that the rule should be in machine-readable form (an assertion inside the member function). For example:
```c++
bool is_versioned() const
{
	const bool is = !(~prtype & DATA_VERSIONED);
	ut_ad(!is || mtype != DATA_SYS);
	return is;
}
```

Reply:
Just tried to reflect that vers_start/end are not included (because I always have to do bitwise arithmetics in my mind to deduce what `is_versioned()` means). I'll try to rephrase better.

Reply:
Actually, by looking into context it's easy to guess what `user field` means while current comment doesn't help anything.

Reply:
"this" refers to `dict_col_t`, which stands for a table column.
While the SQL layer may call table columns "fields" and index fields "key_parts", I do not think that we should introduce confusing terminology into InnoDB. I would accept the following comment:
`/** @return whether this is a system versioned user column */`